packages/presentation/src/icons/Icon.tsx

```diff
@@ -23,16 +25,12 @@ const pxDimensionsBySize: { [size: string]: string } = {
   extraLarge: '40px',
 };
 
-const throwInvalidIconError = (name: string) => {
+const throwInvalidIconError = (name: string | undefined) => {
   throw new Error(`No icon with the name ${name} exists`);
 };
 
-export const Icon = memo(({ name, appearance, size, color, className }: IIconProps) => {
-  const Component = iconsByName[name];
-
-  if (!Component) {
-    throwInvalidIconError(name);
-  }
+export const Icon = memo(({ name, reactComponent, appearance, size, color, className }: IIconProps) => {
+  const Component = name ? iconsByName[name] : reactComponent || throwInvalidIconError(name);
```

Comment:
Maybe a different error if reactComponent is null, and then leave throwInvalidIconError above the same, where it requires a name?